src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -274,25 +449,45 @@ private static string GetFullyQualifiedNamespaceName(INamespaceSymbol symbol, Di
                     continue;
                 }
 
-                foreach (var methodName in methodNames)
+                foreach (var (methodName, receiverTypeName) in methodNames)
                 {
+                    cancellationToken.ThrowIfCancellationRequested();
+
                     var methodSymbols = containerSymbol.GetMembers(methodName).OfType<IMethodSymbol>();
 
                     foreach (var methodSymbol in methodSymbols)
                     {
                         counter.TotalExtensionMethodsChecked++;
 
-                        if (methodSymbol.IsExtensionMethod &&
-                            IsAccessible(methodSymbol, internalsVisible))
+                        if (MatchExtensionMethod(methodSymbol, receiverTypeName, internalsVisible, out var receiverType))
                         {
                             // Find a potential match.
-                            builder.Add(methodSymbol);
+                            builder.Add(receiverType!, methodSymbol);
                         }
                     }
                 }
             }
 
-            return builder.ToImmutable();
+            return builder;
+
+            static bool MatchExtensionMethod(IMethodSymbol method, string filterReceiverTypeName, bool internalsVisible, out ITypeSymbol? receiverType)
+            {
+                receiverType = null;
+                if (!method.IsExtensionMethod || method.Parameters.IsEmpty || !IsAccessible(method, internalsVisible))
+                {
+                    return false;
+                }
+
+                // We get a match if the receiver type name match. 
+                // For complex type, we would check if it matches with filter on whether it's an array.
+                if (filterReceiverTypeName.Length > 0 && !string.Equals(filterReceiverTypeName, GetReceiverTypeName(method.Parameters[0].Type), StringComparison.Ordinal))
```

Comment:
does vb need case insensitive compare?

Reply:
this is metadata name comparison, so I guess no?

Reply:
reasonable enuogh for me.  can always see if someone complains.